actix-files/src/service.rs

```diff
@@ -83,10 +83,10 @@ impl Service<ServiceRequest> for FilesService {
             };
 
         // full file path
-        let path = match self.directory.join(&real_path).canonicalize() {
-            Ok(path) => path,
-            Err(err) => return Box::pin(self.handle_err(err, req)),
-        };
+        let path = self.directory.join(&real_path);
+        if let Err(err) = path.canonicalize() {
+            return Box::pin(self.handle_err(err, req));
+        }
```

Comment:
My concern about this is that we probably want to normalize the path in the way that canonicalize does just without following symlinks. Have you done any testing with paths that includes '../' and such? I'm not sure we have test coverage for it.

Reply:
I think `real_path` is already normalized by `PathBufWrap::parse_path()` and `self.directory` is also a canonical path (see `Files::new()`) , so I think this simple join should pose no problem.

I've just tested it with '../' in both url path and `self.directory`. If you wish I can modify the testcase for this?

Alternatively, we can keep the old code and use `NamedFile::from_file()` to pass the correct filename.

Reply:
It's okay then, I don't see any immediate problems with the change made.